imagetest/test_suites/image_boot/image_boot_test.go

```diff
@@ -123,5 +197,49 @@ func TestBootTime(t *testing.T) {
 	if err != nil {
 		t.Fatalf("failed to convet start time %s", metadata)
 	}
-	t.Logf("image boot time is %d", time.Now().Second()-startTime)
+	t.Logf("image start time is %d", time.Now().Second()-startTime)
+}
+
+func TestBootTime(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	MAX_THRESHOLD := getThresholdValue(image)
+
+	if MAX_THRESHOLD == 0 {
+		t.Fatalf("unrecognized image, no threshold value found")
+	}
+
+	var foundGuestAgentAndSshd bool
+
+	for i := 0; i < 120; i++ {
```

Comment:
Where's the value 120 coming from? is this the number of retries? if so why 120?

Reply:
This is based on the highest value of the all thresholds so the maximum of all is 120 so the value is set to 120 here.

Reply:
can you add a comment explaining this reasoning?

Reply:
So the value 120 is based on the maximum of all thresholds of all images so from the list it is sles-15. This is because we are taking into consideration the worst case for both sshd and guest agent to start for all images. The values are decided by taking enough samples by running the test multiple times. I can update the iteration to run based on each image's expected threshold value.  Also, the lookForProcesses() function it will return an error if both sshd and guest-agent are not found in the process filesystem in that case we wait for a delay of one second and check again giving the process the time to start. Currently we check for it 120 times (120 seconds + time to check (which is minimal in so ignoring)). So if for example for sles-15 image if we not able to find sshd and guest-agent in 120s that means it is taking longer than expected for these processes to starts and violates our expected boot-time conditions so the test fails

Reply:
i do understand :) i suggest a comment on line 216 saying '120 is the current maximum number of seconds of any distro' or similar, so anyone reading the code understands where the number 120 came from